packages/attribute-registry-process/src/services/readModelService.ts

```diff
@@ -1,5 +1,140 @@
-/* import { logger, ReadModelRepository } from "pagopa-interop-commons";
+import { Filter } from "mongodb";
+import { z } from "zod";
+import { logger, ReadModelRepository } from "pagopa-interop-commons";
+import {
+  AttributeKind,
+  Attribute,
+  WithMetadata,
+  genericError,
+  ListResult,
+} from "pagopa-interop-models";
 import { config } from "../utilities/config.js";
-const { attributes } = ReadModelRepository.init(config); */
 
-// TODO
+const { attributes } = ReadModelRepository.init(config);
+
+async function getAttribute(
+  filter: Filter<{ data: Attribute }>
+): Promise<WithMetadata<Attribute> | undefined> {
+  const data = await attributes.findOne(filter, {
+    projection: { data: true, metadata: true },
+  });
+  if (data) {
+    const result = z
+      .object({
+        metadata: z.object({ version: z.number() }),
+        data: Attribute,
+      })
+      .safeParse(data);
+    if (!result.success) {
+      logger.error(
+        `Unable to parse attribute item: result ${JSON.stringify(
+          result
+        )} - data ${JSON.stringify(data)} `
+      );
+      throw genericError("Unable to parse attributes items");
+    }
+    return {
+      data: result.data.data,
+      metadata: { version: result.data.metadata.version },
+    };
+  }
+  return undefined;
+}
```

Comment:
Same as the review I did the other day, I would exit the function early in case there is not data
```suggestion
  const data = await attributes.findOne(filter, {
    projection: { data: true, metadata: true },
  });
  if (!data) {
    return undefined;
  }
    const result = z
      .object({
        metadata: z.object({ version: z.number() }),
        data: Attribute,
      })
      .safeParse(data);
    if (!result.success) {
      logger.error(
        `Unable to parse attribute item: result ${JSON.stringify(
          result
        )} - data ${JSON.stringify(data)} `
      );
      throw genericError("Unable to parse attributes items");
    }
    return {
      data: result.data.data,
      metadata: { version: result.data.metadata.version },
    };}
```